truffle/contracts/VaultNotary/VaultNotary.sol

```diff
@@ -0,0 +1,243 @@
+        _;
+    }
+
+    /** @notice This modifier is for testing whether a Vault has been registered yet
+      * @param vaultId bytes16 The ID of the vault to check
+      */
+    modifier isNotRegistered(bytes16 vaultId) {
+        require(notaryMapping[vaultId].vaultOwner == address(0x0), "Vault ID already exists");
+        _;
+    }
+
+    /** @notice This modifier is for testing whether a Vault has been registered yet
+      * @param vaultId bytes16 The ID of the vault to check
+      */
+    modifier isRegistered(bytes16 vaultId) {
+        require(notaryMapping[vaultId].vaultOwner != address(0x0), "Vault ID does not exist");
+        _;
+    }
+
+    /** @notice If we upgrade the version of the contract, the old contract
+      *  will not be allowed to register new Vault anymore. However, the existing
+      *  Vaults registered in that contract should still be able to be updated
+      */
+    modifier notDeprecated() {
+        require(!isDeprecated, "This version of the VaultNotary contract has been deprecated");
+        _;
+    }
+
+    /** @notice This is function to register a vault, will only do the registration if a vaultId has not
+                been registered before
+      * @dev It sets the msg.sender to the vault owner, and calls the setVaultUri and setVaultHash to
+      * initialize the uri and hash of a vault
+      * @param vaultId bytes16 VaultID to create, is the same as shipment ID in our system
+      * @param vaultUri string Vault URI to set
+      * @param vaultHash string  Vault hash to set
+      */
+    function registerVault(bytes16 vaultId, string calldata vaultUri, string calldata vaultHash)
+        external
+        isNotRegistered(vaultId)
+        notDeprecated
+    {
+        notaryMapping[vaultId].vaultOwner = msg.sender;
+
+        //work around for if (vaultUri != "")
+        if (bytes(vaultUri).length != 0)
+            setVaultUri(vaultId, vaultUri);
+
+        if (bytes(vaultHash).length != 0)
+            setVaultHash(vaultId, vaultHash);
+
+        emit VaultRegistered(msg.sender, vaultId);
+    }
+
+    /** @notice Sets the contract isDeprecated flag. Vault registration will be disabled if isDeprecated == True
+      * Only contract owner can set this
+      * @param _isDeprecated bool Boolean control variable
+      */
+    function setDeprecated(bool _isDeprecated)
+        external
+        onlyOwner
+    {
+        isDeprecated = _isDeprecated;
+        emit ContractDeprecatedSet(msg.sender, isDeprecated);
+    }
+
+    /** @notice Function to grant update permission to both Hash field in one vault
+      * @param vaultId bytes16 The ID of Vault to grant permission
+      * @param addressToGrant address The address to grant permission
+      */
+    function grantUpdateHashPermission(bytes16 vaultId, address addressToGrant)
+        external
+        vaultOwnerOnly(vaultId)
+        isRegistered(vaultId)
```

Comment:
For the functions where we include multiple modifiers, we should order them from least specific to most specific.  For example in this function, `isRegistered` should come before `vaultOwnerOnly`.  Several of these functions have this same incorrect ordering.  Please review the ordering of modifiers on all methods.